### ISSUE TITLE
NVDA is not reading expanded/collapsed on menuitem with aria-haspopup as "true"

### ISSUE THREAD

**Issue by JamesBurnside** [nvaccess/nvda#18479]
### Steps to reproduce:

Create a menuitem with `aria-expanded="true/false"` attribute and `aria-haspopup="true"`

See codepen repro: https://codepen.io/jaburnsi/pen/zxvGbVo, item 1 does not announce collapsed state as it has `aria-haspopup="true"`, whereas item2 does correctly announce collapsed state 

### Actual behavior:

<img width="495" height="488" alt="Image" src="https://github.com/user-attachments/assets/62edd365-1fbc-4a43-b786-e7c740ff0579" />

### Expected behavior:

Should say collapsed state similar to when it doesn't have aria-popup=true:

<img width="483" height="492" alt="Image" src="https://github.com/user-attachments/assets/10ffaa41-9e19-4b46-99ce-0fd745f54680" />

#### NVDA installed/portable/running from source:

Launcher > Continue Running

#### NVDA version:

2025.1.2.36913

#### Windows version:

Win11 24H2 (26100.4652)

#### Name and version of other software in use when reproducing the issue:

Edge Chromium 138.0.3351.77 

and

Firefox 140.0.4 (64-bit)

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

#### If NVDA add-ons are disabled, is your problem still occurring?

Not tested, running default configuration.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Not tested.

**Comment by JamesBurnside** [comment]
Note: this appears to be the same issue as #9096 that has been fixed, except this issue is for `menuitem` and that issue refers to `button`

**Comment by Adriani90** [comment]
It seems this is reproducible in every browser, Chrome and Fiorefox.
It seems NVDA ignores aria-expanded atribute if aria-has-popup has been set on the menuitem.
Actually NVDA should report:
"menuitem colapsed submenu item 1"
according to Aria spec.
cc: @jcsteh since it also affects Firefox.

**Comment by jcsteh** [comment]
This is intentional. It isn't usually useful to know if a menuitem which has a sub-menu is expanded or collapsed, since you shouldn't be able to focus a menuitem which is expanded, and if it's collapsed, the fact that it has a sub-menu is sufficient indication that it can be expanded. So, this was done to reduce unnecessary verbosity.

Note that if you report focus with NVDA+tab or report the current navigator object (e.g. NVDA+numpad5), NVDA will report this information.

What's the real world use case for this being important?

> according to Aria spec.

The ARIA spec doesn't explicitly specify how information should be reported by screen readers. While screen readers should generally report all the available information, there are cases like these where a screen reader may choose to eliminate or abbreviate in order to make things more efficient for the user. As a simple example, the ARIA state requires exposure of the focusable state on all focusable items, but while screen readers do use that information, they don't report it, since hearing "focusable" on every control just isn't useful.

**Comment by Adriani90** [comment]
Thanks Jamie for the clarification.
I agree users already know that a menu is colapsed when they hear "submenu", this is the indication that you can open a submenu by pressing right arrow or space bar etc.
Jaws indeed reports "menu submenu colapsed", which is quite verbose in my opinion.
Given the behavior in NVDA is by design and the user gets the same information when hearing "submenu", I am closing as won't fix.
Please comment if you have any use case or arguments showing why hearing "colapsed / expanded" on submenus is relevant for the user, and we can reopen the discussion.

**Comment by jcsteh** [comment]
For reference, the relevant code is https://github.com/nvaccess/nvda/blob/d05da569885ff668b075c7dcffe92757d87f3442/source/controlTypes/processAndLabelStates.py#LL83
